Automattic/pocket-casts-ios
# Tracks: Player sleep timer events

| 📘 Project: #154 | 🛫 Depends on: #289 |
|:---:|:---:|

This adds events for the sleep timer:

- `player_sleep_timer_enabled`: When the sleep timer is enabled
- `player_sleep_timer_extended`: When the user extends the sleep timer duration
- `player_sleep_timer_cancelled`: When the user cancels the sleep timer

## To test

1. Enable the tracksEnabled option in FeatureFlag.swift
2. Play an episode if you don't have one playing already
3. Open the full screen player
4. Tap the ZZZ icon to show the sleep timer options
5. Tap the 5 minutes option
6. ✅ Verify you see: `🔵 Tracked: player_sleep_timer_enabled ["time": 300]` in console
7. Tap each of the other items except end of episode
8. ✅ Verify the `time` property in the log corresponds with the item you selected
9. Tap the end of episode item
10. ✅ Verify you see `🔵 Tracked: player_sleep_timer_enabled ["time": "end_of_episode"]` in console 
11. Tap the Zzz icon again and tap then cancel button
12. ✅  `🔵 Tracked: player_sleep_timer_cancelled`
13. Enable the 5 minute sleep timer
14. Tap the ZZZ icon again
15. Tap the `+ 5 Minutes` button
16. ✅ `🔵 Tracked: player_sleep_timer_extended ["amount": 300]`
17. Tap the End of Episode button
18. ✅ Verify you see `🔵 Tracked: player_sleep_timer_extended ["amount": "end_of_episode"]`
19. Tap cancel again

## Checklist
